### ISSUE TITLE
add_overall() breaks when columns are added with by_values

### ISSUE THREAD

**Issue by tgerke** [shannonpileggi/gtreg#122]
Sometimes I need to add a column with zero counts using`by_values` for example if there were no "Grade 2" AEs. `by_values` does nicely, except when you attempt to run the results through `add_overall()`. Reprex below.

```r
# works without add_overall()
df_adverse_events %>%
  tbl_ae(
    id = patient_id,
    ae = adverse_event,
    soc = system_organ_class, 
    by = grade,
    by_values = as.character(0:5)
  )

# works without by_values
df_adverse_events %>%
  tbl_ae(
    id = patient_id,
    ae = adverse_event,
    soc = system_organ_class, 
    by = grade
  ) %>% 
  add_overall()

# throws error
df_adverse_events %>%
  tbl_ae(
    id = patient_id,
    ae = adverse_event,
    soc = system_organ_class, 
    by = grade,
    by_values = as.character(0:5)
  ) %>%
  add_overall()
#> Error: Cannot specify `by_values=` without also specifying `by=`.
```

<sup>Created on 2022-03-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by shannonpileggi** [comment]
I confirmed that I see the same thing when I run from the version in main.

I know we saw this before, but not sure where we addressed. @ddsjoberg I looked through our chat and we discussed on Jan 14, I wonder if it was in [Bad Memories](https://github.com/shannonpileggi/gtreg/pull/74) 😱? Not sure if we tested w/ `add_overall()` though.

**Comment by shannonpileggi** [comment]
@tgerke, until we get the bug fixed you can convert grade to a factor before you pass to `tbl_ae()` - this works.

```
df_adverse_events %>%
  mutate(grade_fct = fct_expand(factor(grade), "0") %>% 
         fct_relevel("0", "1", "2", "3", "4", "5")) %>% 
  tbl_ae(
    id = patient_id,
    ae = adverse_event,
    soc = system_organ_class, 
    by = grade
  ) %>%
  add_overall()
```

![image](https://user-images.githubusercontent.com/17747575/158494473-4f7fcd5b-9872-4089-9f06-683e0bcbb366.png)

**Comment by ddsjoberg** [comment]
Curious...I thought we made this exact update previously...hmmm.

Regardless, fix re-implemented in PR #123
